### ISSUE TITLE
Learning with ICM 

### ISSUE THREAD

**Issue by zafarmah92** [simoninithomas/Baselines_icm#1]
Hi, 

Many thanks for embedding the network, indeed its very helpful. I am also working with this model together with OpenAI baselines, I do have embedded this A2C but the parallel information create some problem for the learning process as episode reward after 2e3 timesteps completely declines.  I would be glad if you can give me some thoughts 

1. **Batch processing by each worker** : while returning the forward loss from ICM file in 
```
 self.forw_loss = tf.reduce_mean(tf.square(tf.subtract(pred_phi_next_state, phi_next_state)), axis=-1,name="forward_loss")

```
Is it necessary to provide axis here ? giving the axis would return each independent worker forward loss in the format of an array which is used as **mb_rewards** from processing
 
2. **Training of ICM** : 
As I know that this work is still in progress but wants to ask, isn't it good to use single optimizer function for ICM and PPO as done in the [pathak](https://github.com/pathak22/noreward-rl) code. where to combine both (policy optimizer and ICM) parametric information. 
If not I would be glad if you can through some light and look forward to remaining function.
